src/util/isoDateTime/isoDateTime.ts

```diff
@@ -0,0 +1,15 @@
+const getYearFromISODateTime = (ISODateTime: string, locale: string): string => {
+  const date = new Date(ISODateTime)
+  const formatter = new Intl.DateTimeFormat(locale, { year: 'numeric' })
+  return formatter.format(date)
+}
```

Comment:
In the function, I would delegate the responsibility of determining whether a string is an ISO DateTime to the `Date` constructor. Moreover, as it stands, if you pass a string that does not comply with the format, the function throws an exception.

Reply:
Yes, let's delegate the responsibility to the Date constructor and treat returned value (or Error) before calling `convertToYearIfISODateTime` method: it will prevent from returning a non formatted input.

Reply:
What do you think about a validation that use both the regex and the `Date` constructor to determine if a string is an ISODateTime? Because for example `new Date(1)` is a date.

> Moreover, as it stands, if you pass a string that does not comply with the format, the function throws an exception.
You are right, that the reason why I didn't exported the function. That's probably too implicit and needs error handling if we want to use it directly.

Reply:
Yes. My previous comment was inaccurate. You should rely on the `Date.parse()` function to do what you want without needing a regex. Just check that the number returned is not `NaN`, and it will work as expected.